app/controllers/WhatDoYouWantToDoController.scala

```diff
@@ -120,12 +118,18 @@ class WhatDoYouWantToDoController @Inject()(
 
   private def allowWhatDoYouWantToDo(implicit request: Request[AnyContent], user: AuthedUser): Future[Result] = {
     val nino = user.nino
-    for {
-      hasTaxCodeChanged <- taxCodeChangeService.hasTaxCodeChanged(nino)
-      showJrsTile       <- jrsService.checkIfJrsClaimsDataExist(nino)
-      model             <- whatToDoView(nino, hasTaxCodeChanged, showJrsTile)
-      _                 <- auditNumberOfTaxCodesReturned(nino, showJrsTile)
-    } yield Ok(whatDoYouWantToDoTileView(WhatDoYouWantToDoForm.createForm, model, applicationConfig))
+
+    taxCodeChangeService.hasTaxCodeChanged(nino) flatMap {
+      case Right(taxCodeChanged) =>
+        for {
+          showJrsTile <- jrsService.checkIfJrsClaimsDataExist(nino)
+          model       <- whatToDoView(nino, taxCodeChanged, showJrsTile)
```

Comment:
no big deal but whatToDoView and auditNumberOfTax can be done in parallel: 
```
import cats.implicits._
(model, _)       <- (whatToDoView(nino, taxCodeChanged, showJrsTile), auditNumberOfTaxCodesReturned(nino, showJrsTile)).tupled
```

Reply:
We can do that 👍